### ISSUE TITLE
decompiler: recognise float sign changes via and/or/xor bit manipulation

### ISSUE THREAD

**Issue by slippycheeze** [NationalSecurityAgency/ghidra#5785]
**Is your feature request related to a problem? Please describe.**
MSVC, in the X84 64-bit project I'm working on, generates bitmask operations to manipulate float32 sign for implementing "force float to be positive" and "force float to be negative" operations.

An example of the the code output is:
```c++
// fVar8 is confirmed to be a genuine float type.
if (fVar8 >= *(float *)(param_1 + 0x20)) {
    fVar6 = (float)((uint)fVar6 ^ 0x80000000);
}
```

The strictly related X86-64 assembly code follows; instructions not strictly related cropped out.
```
180116dd8  MOVSS  XMM1, dword ptr [RCX + this->unknown_purpose_int
180116dec  MULSS  XMM1, dword ptr [DAT_18034a868]     ;  = C5h
180116dff  MOVSS  XMM4, dword ptr [DAT_18034b1e0]     ;  = 80000000h
180116e0e  XORPS  XMM1, XMM4
```

The result is to flip the sign on the float32, by XOR-ing the sign bit.  Similar things for forcing a float32  to be positive or negative, with `AND 0x80000000` or `AND ~0x80000000` elsewhere.   Because of how float32 is defined, this has no other effect on the value, so is a reasonable and efficient way to optimize those operations.

NOTE: I have at least one instance where this technique was applied to `XMM1 <=> float[4]` in the code, and the source data for the `XOR/OR/AND` was `uint[4]` of 4 `dw 80000000h` values.  so the "vectorized" version of this also applies.

**Describe the solution you'd like**
I'd like to see this decompiler output, or at least, to "not see" the `(float)((uint)fVar6` casting...
```c++
fVar6 = -fVar6;                       // XOR   0x80000000 ⇒ unary minus
fVar6 = __intrinsic_fabsf(fVar6);     // AND ~ 0x80000000 ⇒ "make this absolute" as a function call
fVar6 = - __instrinsic_fabsf(fVar6);  // AND   0x80000000 ⇒ unary minus and fabsf, least worst choice
```

...or something similar.  even this would work for me better than the current output with the (accurate, but unhelpful) casting through `uint` on the way to our sign-mangled float32.
```c++
if (fVar6 < 0.0) { fVar6 = -fVar6; }
```

**Additional context**
Debug Decompiler Output XML for a function demonstrating this; scan for `XORPS XMM1, XMM4` at address `180116e0e` in the function.  that is `0x7E` bytes from the function entry point; `XMM4` has been loaded from constant defined data containing `dw 80000000h`, to provide the mask, XMM1 has a genuine float in the lowest 32 bits.

[decompiler-feature-x64-64-float32-bitmask-for-sign-manipulation.zip](https://github.com/NationalSecurityAgency/ghidra/files/12640389/decompiler-feature-x64-64-float32-bitmask-for-sign-manipulation.zip)

**Comment by intrigus-lgtm** [comment]
This looks a bit like a duplicate of #515

**Comment by caheckman** [comment]
Fixed by 520dc99b1119f5bd869f0e590725f2f1ec5ded72
